dask_ml/model_selection/_incremental.py

```diff
@@ -654,13 +655,15 @@ def _get_params(self):
             return ParameterSampler(self.parameters, self.n_initial_parameters)
 
     def _additional_calls(self, info):
-        best = self._adapt(info)
-        bad = set(info) - set(best)
+        self._all_models.update(set(info.keys()))
+        instructions = self._adapt(info)
+        bad = set(self._all_models) - set(instructions)
```

Comment:
I'm getting a bug with this sometimes. This commit attempts to resolve it. On occasion, I get a `KeyError` in `_incremental.py` with `speculative.pop(ident)` at https://github.com/dask/dask-ml/blob/37eca7d335509c2a4aa9332aa454f57092318487/dask_ml/model_selection/_incremental.py#L230  

which I take to mean some models are being killed early. This happens `bad == set()` at https://github.com/dask/dask-ml/blob/37eca7d335509c2a4aa9332aa454f57092318487/dask_ml/model_selection/_incremental.py#L212

Reply:
@mrocklin any thoughts here? I've seen this occasionally to, but haven't been able to debug it successfully.

Reply:
No thoughts off-hand.  Is anyone able to produce a minimal example by any chance?